src/Raven.Server/Documents/ResourceCache.cs

```diff
@@ -72,8 +72,23 @@ private bool UnlikelyTryGet(StringSegment resourceName, out Task<TResource> reso
             if (_caseInsensitive.TryGetValue(resourceName, out resourceTask) == false)
                 return false;
 
+            _forTestingPurposes?.OnUnlikelyTryGet?.Invoke();
+
             lock (this)
             {
+                if (_caseInsensitive.TryGetValue(resourceName, out var resourceTaskUnderLock) == false)
+                {
+                    // database was deleted
+                    return false;
```

Comment:
When I debug the added test locally then it always ended up in this check. Is it really the test case scenario that a database was deleted?

Reply:
We talked about that directly. The answer is yes - what we do in the test is the same what we do during db deletion